R/sendData.R

```diff
@@ -142,35 +113,67 @@ writeTable <- function(db, path, table, names, mode) {
 }
 
 sqlLoadData <- function(path, table, names, mode) {
-  SingleQuery (
-    paste0 (
-      "LOAD DATA LOCAL INFILE '",
-      path,
-      "' ",
-      if (mode == "replace")
-        "REPLACE ",
-      "INTO TABLE `",
-      table,
-      "` ",
-      "CHARACTER SET UTF8 ",
-      "FIELDS TERMINATED BY ',' ",
-      "OPTIONALLY ENCLOSED BY '\"' ",
-      "LINES TERMINATED BY '\n' ",
-      "IGNORE 1 LINES ",
-      sqlParan(sqlEsc(names)),
-      ";"
+  SingleQuery(
+    paste0(
+      "load data local infile '", path, "' ",
+      if (mode == "replace") "replace ",
+      "into table ", sqlEsc(table), " ",
+      "character set utf8 ",
+      "fields terminated by ',' ",
+      "optionally enclosed by '\"' ",
+      "lines terminated by '\n' ",
+      "ignore 1 lines ",
+      sqlParan(sqlEsc(names)), ";"
     )
   )
 }
 
-sqlUpdateData <- function(table, cols, values, colsUpdate) {
-  paste(
-    sep = " ",
-    "INSERT INTO", table,
-    cols,
-    "VALUES", sqlParan(values),
-    "ON DUPLICATE KEY UPDATE",
-    colsUpdate,
-    ";"
+updateTable <- function(db, path, table, names) {
+
+  # 1. create temporary table like target table
+  createTemporaryTable(db, table)
```

Comment:
Das funktioniert so noch nicht in folgendem Szenario: Wir haben eine Tabelle mit Spalten 'key', 'a' und 'b'. Sei für key = 1, a = 2 und b = 3 in der Datenbank. Wenn wir nun ein Update ausführen mit den Daten: `data.frame(key = 1, a = 3)`, dann sind die Daten in der Datenbank danach: key = 1, a = 3 und b = NULL. Erwarten würden wir aber key = 1, a = 3 und b = 3.

- Wir brauchen das als Unit-Test
- Der Create Temporary Table muss angepasst werden. 
    - Entweder wir erzeugen nur die Spalten die in den Daten für das Update benötigt werden, 
    - oder wir löschen die Spalten die nicht benötigt werden.

Ich vermute Spalten zu löschen ist einfacher umzusetzen, allerdings braucht der User immer mehr Rechte auf der Datenbank. Das Create Temporary Table Statement anzupassen ist also sauberer und sparsamer in Bezug auf die Rechtevergabe.

Reply:
> create temporary table like target table

müsste vermutlich 

> create temporary table like cached table

sein